docs/refs/configuring_mbed_os.md

```diff
@@ -1,13 +1,17 @@
+
+All threads in Mbed OS share a global heap. By default, memory for thread stacks is dynamically allocated from the global heap. User spawned thread stacks can be allocated from other memory areas. The size of the global heap depends on the compiler. The heap size is static when compiling with IAR, and it is different for each target in Mbed OS. You can find it in the IAR linker configuration files nested inside target directories. For example, you can find the linker configuration file for a K66F [here](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_K66F/device/TOOLCHAIN_IAR/MK66FN2M0xxx18.icf#L49-L51). For the GCC and ARM compilers, the heap size is dynamic based on RAM usage.
 
 #### Deafult threads
-By default, there are four threads running after boot: the ISR/scheduler thread, the idle thread, the timer thread, and the main application thread. Combined, the idle thread, timer thread, and ISR/scheduler thread consume 2 kilobytes of RAM. These cannot be modified by the user. The size of the main application thread stack can be expanded or reduced using the mbed configuration system.
 
-#### Configuring main thread stack
-The default stack size of main application thread is 4 kilobytes. This memory is dynamically allocated from the global heap. The stack size of the main thread can be configured using the [mbed config system](https://docs.mbed.com/docs/mbed-os-handbook/en/latest/advanced/config_system/). To reduce the stack size from 4K to 2K, create an `mbed_app.json` file for your project with the following content:
+By default, there are four threads running after boot: the ISR/scheduler thread, the idle thread, the timer thread and the main application thread. Combined, the idle thread, timer thread and ISR/scheduler thread consume 2 kilobytes of RAM. The user cannot modify these. You can expand or reduce the size of the main application thread stack by using the Mbed configuration system.
+
+#### Configuring the main thread stack
+
+The default stack size of the main application thread is 4 kilobytes. This memory is dynamically allocated from the global heap. You can configure the stack size of the main thread by using the [Mbed config system](https://docs.mbed.com/docs/mbed-os-handbook/en/latest/advanced/config_system/). To reduce the stack size from 4K to 2K, create an `mbed_app.json` file for your project with the following content:
```

Comment:
Query: Would it still be correct to change the second sentence to "The global heap dynamically allocates this memory." for active voice?

Reply:
I think it might be more accurate to say "mbed OS dynamically allocates this memory from the global heap." Is that better?

Reply:
https://github.com/sarahmarshy/Handbook/commit/abd50896392cd98cf9036a78097db0106d88b97b